src/server/admin/server/api_server.go

```diff
@@ -170,6 +170,11 @@ func (a *apiServer) Extract(request *admin.ExtractRequest, extractServer admin.A
 			if ci.ParentCommit == nil {
 				ci.ParentCommit = client.NewCommit(ci.Commit.Repo.Name, "")
 			}
+			// Restore must not create any open commits (which can interfere with
+			// restoring other commits), so started and finished are always set
+			if ci.Finished == nil {
```

Comment:
What happens if you have call extract on a cluster that has legitimately open commits that the user just hasn't closed yet. Is this possible, and if so would it be good to warn the user like "hey, you tried to extract a cluster with open commits, we closed them for you." or something?

Reply:
Per face-to-face discussion, I'm happy to add a warning. My current feeling is that it's fine—I think this would come up if extract/restore ran while a user was loading data, which would be difficult to make work (we'd have to extract data as they were loading it)